Incorrect disk corruption check when `No space left on device`

<img width="1760" height="260" alt="Image" src="https://github.com/user-attachments/assets/e1a5a914-4c0a-4bf2-b842-b20d29979b62" />

This will make disk as corruption state and then make the corresponding spark apps fast fail.


client log as follows:

```
26/02/03 10:22:37 ERROR ShuffleServerGrpcNettyClient: Can't get shuffle data from x.x.x.x:29104 for appId[application_1769588986968_1913800_1770068667871], shuffleId[4], partitionId[603], errorMsg:Errors on getting file data. err: LOCAL_DISK_OWNED_BY_PARTITION_CORRUPTED(
    "/data4/riffle/data/v0.18.0-rc1",
)
26/02/03 10:22:37 ERROR PrefetchableClientReadHandler: Errors on doing readShuffleData
org.apache.uniffle.common.exception.RssException: Failed to read shuffle data with ClientInfo{clientType=GRPC_NETTY, shuffleServerInfo=ShuffleServerInfo{host[x.x.x.x], grpc port[29100], netty port[29104]}}
	at org.apache.uniffle.storage.handler.impl.LocalFileClientReadHandler.readShuffleData(LocalFileClientReadHandler.java:207)
	at org.apache.uniffle.storage.handler.impl.DataSkippableReadHandler.doReadShuffleData(DataSkippableReadHandler.java:111)
	at org.apache.uniffle.storage.handler.impl.PrefetchableClientReadHandler.lambda$readShuffleData$0(PrefetchableClientReadHandler.java:103)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.uniffle.common.exception.RssFetchFailedException: Can't get shuffle data from x.x.x.x:29104 for appId[application_1769588986968_1913800_1770068667871], shuffleId[4], partitionId[603], errorMsg:Errors on getting file data. err: LOCAL_DISK_OWNED_BY_PARTITION_CORRUPTED(
    "/data4/riffle/data/v0.18.0-rc1",
)
	at org.apache.uniffle.client.impl.grpc.ShuffleServerGrpcNettyClient.getShuffleData(ShuffleServerGrpcNettyClient.java:485)
	at org.apache.uniffle.storage.handler.impl.LocalFileClientReadHandler.readShuffleData(LocalFileClientReadHandler.java:193)
	... 7 more
```

--- a/riffle-server/src/store/local/delegator.rs
+++ b/riffle-server/src/store/local/delegator.rs
@@ -229,11 +229,25 @@ impl LocalDiskDelegator {
                         self.mark_operation_abnormal();
                     }
                     error => {
-                        error!(
-                            "Errors on checking the disk:{} write+read. err: {:#?}",
-                            &self.inner.root, error
-                        );
-                        self.mark_corrupted()?;
+                        // Check if this is an ENOSPC (no space left on device) error
+                        let error_msg = error.to_string();
+                        let is_enospc = error_msg.contains("No space left on device")
+                            || error_msg.contains("ENOSPC")
+                            || error_msg.contains("Storage full");
+
+                        if is_enospc {
+                            warn!(
+                                "Disk {} has insufficient space. err: {:#?}",
+                                &self.inner.root, error
+                            );
+                            self.mark_space_insufficient();
+                        } else {
+                            error!(
+                                "Errors on checking the disk:{} write+read. err: {:#?}",
+                                &self.inner.root, error
+                            );
+                            self.mark_corrupted()?;
+                        }
                     }
                 },
                 Ok(_) => {

